Zotero/Scenes/Detail/SingleCitation/Views/SingleCitationViewController.swift

```diff
@@ -46,95 +46,144 @@ final class SingleCitationViewController: UIViewController {
+        omitAuthorSwitch.setOn(viewModel.state.omitAuthor, animated: false)
+        setupPreview()
+        setupNavigationBar()
+        setupObserving()
 
-        self.viewModel.process(action: .preload(self.previewWebView))
-    }
+        viewModel.process(action: .preload(previewWebView))
 
-    override func viewWillAppear(_ animated: Bool) {
-        super.viewWillAppear(animated)
-        self.updatePreferredContentSize()
-    }
+        func setupPreview() {
+            previewContainer.layer.cornerRadius = 4
+            previewContainer.layer.masksToBounds = true
 
-    deinit {
-        self.viewModel.process(action: .cleanup)
-    }
-
-    // MARK: - Actions
+            switch UIDevice.current.userInterfaceIdiom {
+            case .pad:
+                view.safeAreaLayoutGuide.bottomAnchor.constraint(equalTo: container.bottomAnchor, constant: 12).isActive = true
 
-    private func update(state: SingleCitationState) {
-        if state.changes.contains(.locator) {
-            self.locatorButton.setTitle(self.localized(locator: state.locator), for: .normal)
-        }
+            case .phone:
+                view.safeAreaLayoutGuide.bottomAnchor.constraint(greaterThanOrEqualTo: container.bottomAnchor, constant: 12).isActive = true
 
-        self.updatePreview(isLoading: state.loadingPreview)
-        self.setupRightButtonItem(isLoading: state.loadingCopy)
-        self.navigationItem.rightBarButtonItem?.isEnabled = !state.loadingPreview
-
-        if let error = state.error {
-            switch error {
-            case .styleMissing:
-                self.coordinatorDelegate?.showMissingStyleError()
-
-            case .cantPreloadWebView:
-                self.coordinatorDelegate?.showCitationPreview(errorMessage: L10n.Errors.citationPreview)
+            default:
+                break
             }
+
+            previewWebView.scrollView.isScrollEnabled = false
+            previewWebView.backgroundColor = .clear
+            previewWebView.scrollView.backgroundColor = .clear
+            previewWebView.configuration.userContentController.add(self, name: "heightHandler")
         }
 
-        if state.changes.contains(.preview) {
-//            do {
-//                if let data = state.preview.data(using: .utf8) {
-//                    let attr = try NSAttributedString(data: data, options: [.documentType: NSAttributedString.DocumentType.rtf, .characterEncoding: String.Encoding.utf8], documentAttributes: nil)
-//                    self.previewTitleLabel.numberOfLines = 0
-//                    self.previewTitleLabel.attributedText = attr
-//                }
-//            } catch let error {
-//                NSLog("ERROR: \(error)")
-//            }
-            self.updatePreferredContentSize()
+        func setupNavigationBar() {
+            let cancel = UIBarButtonItem(title: L10n.cancel, style: .plain, target: nil, action: nil)
+            cancel.rx.tap.subscribe(onNext: { [weak self] in
```

Comment:
You can change this to `subscribe(with: self, onNext: { self in`

Reply:
Both styles are used throughout the codebase, and as I mentioned in a previous comment, the convenience `subscribe(with: ...)` method may be limiting in certain edge cases. Of course we can opt to not use it only then, and add also a comment in the code, to justify this change.
However, this convenience method, when user with `self`, as it is needed to be differentiated from the `object` parameter, by writing
```
subscribe(with: self, onNext: { `self`, _ in ...
```
means that you cannot allow implicit self for weak self captures, after self is unwrapped.
I.e. if you need to refer to multiple `self` properties you need to be explicit, such as
```
subscribe(with: self, onNext: { `self`, _ in
    self.viewModel.process(action: .setOmitAuthor(self.omitAuthorSwitch.isOn))
})
```
vs.
```
subscribe(onNext: { [weak self] _ in
    guard let self else { return }
    viewModel.process(action: .setOmitAuthor(omitAuthorSwitch.isOn))
})
```
Which direction should we follow?

Reply:
We use both styles because the `with: self` didn't exist initially and I started moving forward to it after it was introduced. But yes now that we use implicit `self` with `guard` it makes more sense to not use this feature. At least we'll have the same style in `RxSwift` and other closures.

So yes, sorry about the confusion, we'll keep doing the `weak` + `guard` dance to be able to use implicit `self` in closures.